### DISCUSSION TITLE
Esp32 Bluetooth proxy

### DISCUSSION THREAD

**Discussion by JoshuaSeidel** [theengs/gateway#279] [Q&A]
Is there a way to configure this to use a esp32 Bluetooth proxy as the adapter? 

I'm looking to use this in home assistant with Bluetooth proxies around my home to track iPhones and Apple Watches to be used with automation

**Comment by DigiH** [comment]
You can use the `esp32dev-ble-mqtt-undecoded` binary of [OpenMQTTGateway](https://docs.openmqttgateway.com/upload/web-install.html#using-openmqttgateway) on one or more ESP32s to cover the reception of your iPhones and Apple Watches in your whole house.

This will give you reliable device tracker Home/Away presence detection.

If you are looking for a room specific detection you might have to look for a different solution.

**Reply by mihsu81** [reply]
@DigiH Considering the popularity of https://github.com/agittins/bermuda maybe it would be a good ideea to collaborate and pass the Bluetooth traffic in a way Bermuda understands it. It would potentially revive the popularity of Theengs which seems to be declining lately.

**Reply by DigiH** [reply]
@mihsu81 Bermuda already seems to be doing that for room specific presence detection, so I'm not quite sure how a collaboration would benefit either side, especially when e`sp32dev-ble-mqtt-undecoded` OpenMQTTGateway passes on undecoded unresolved data. 

Admittedly without me having any practical experience with Bermuda, so open to more details of what you are thinking about. :)

It also looks as if Bermuda is Home Assistant only, while Theengs Gateway is totally controller agnostic, and while the HA Add-on might not be as well known or popular recently, mainly due to exposure I think, as many positive replies show whenever pointed to the TGW Add-on ;) there are controllers where Theengs is the main BLE decoder plug-in, being used with discovery or purely manual.
